node/src/components/consensus/highway_core/state.rs

```diff
@@ -551,6 +559,26 @@ impl<C: Context> State<C> {
                 }
             }
         }
+        // All endorsed units from the panorama of this vote.s
+        let endorsements_in_panorama: Vec<C::Hash> =
+            panorama
+                .iter_correct_hashes()
+                .fold(Vec::new(), |mut acc, hash| {
+                    acc.extend(
+                        self.opt_endorsements(hash)
```

Comment:
Aren't those the endorsements in our local state?
We can't declare a unit invalid because it doesn't claim a unit is endorsed that we locally see as endorsed.

Monotonicity just means that if `hash` claims a unit is endorsed, then `wunit` must also claim that.
So I guess this should be something like `state.unit(hash).endorsement_claims()` (not sure if that method exists yet).

Reply:
Of course. I forgot that `state.endorsed(unit)` may returns endorsements that _this unit_ might have not seen.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/527/commits/ac33bd7178f0bce5cf0a48ded845971daf06b805